sw/host/opentitantool/src/command/rom_patch.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use log::*;
+use serde_annotate::Annotate;
+use std::any::Any;
+use std::io::Read;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::crypto::rsa::RsaPrivateKey;
+use opentitanlib::rom_patch::patch::RomPatchPartition;
+
+/// Generate an OTP ROM patch partition
+#[derive(Debug, StructOpt)]
+pub struct Generate {
+    #[structopt(
+        short = "p",
+        long = "patch",
+        name = "HJSON_FILE",
+        help = "HJSON formatted ROM patch description file"
+    )]
+    patch: PathBuf,
+
+    #[structopt(
+        short = "k", long = "key",
+        name = "DER_FILE",
+        parse(try_from_str=RsaPrivateKey::from_pkcs8_der_file),
+        help = "RSA private key file in PKCS#1 DER format"
+    )]
+    key: RsaPrivateKey,
+
+    #[structopt(
+        short = "s",
+        long = "otp_size",
+        name = "OTP_SIZE",
+        help = "OTP ROM patch partition total size (bytes)"
+    )]
+    otp_size: usize,
+}
+
+impl CommandDispatch for Generate {
+    fn run(
```

Comment:
Can you explain/document what this command is supposed to generate? I was expecting it to generate a binary file (i.e. one that corresponds to the format that you implemented `Read` for). But this seems to generate a C file. What will this C file be used for?

Reply:
We now generate a blob into a file, or an HJSON OTP image partition to be used as an OTP image overlay. I hope it makes more sense.